File: UVa/10450.cpp
#include <iostream>
using namespace std;

int main(){
  long long fib[55];
  fib[0] = 1;
  fib[1] = 1;

  for(int i = 2; i < 55; i++)
    fib[i] = fib[i-2] + fib[i-1];

  int caso, x; cin >> caso;
  for(int i = 1; i <= caso; i++){
    cin >> x;
    cout << "Scenario #" << i << ":" << endl << fib[x+1] << endl << endl;
  }

  return 0;
}

File: UVa/10450.java
import java.util.Scanner;
import java.math.BigInteger;

class Main{
    public static void main(String[] args){
	Scanner scan = new Scanner(System.in);
	BigInteger big[] = new BigInteger[55];

	big[0] = BigInteger.ONE;
	big[1] = BigInteger.ONE;

	for(int i = 2; i < 55; i++)
	    big[i] = big[i-2].add(big[i-1]);

	int caso;
	caso = scan.nextInt();

	for(int i = 1; i <= caso; i++){
	    System.out.print("Scenario #" + i + ":\n" + big[scan.nextInt()+1] + "\n\n");
	}
    }
}
